public/app/features/transformers/calculateHeatmap/heatmap.ts

```diff
@@ -54,7 +55,7 @@ export function bucketsToScanlines(frame: DataFrame): DataFrame {
   const yField = frame.fields[1];
 
   // similar to initBins() below
-  const len = xValues.length * frames.length;
```

Comment:
this is why it broke after the frame-per-bucket -> field-per-bucket changes. (forgot to update here)